main.ts

```diff
@@ -23,7 +25,24 @@ await main(function* () {
 
   const entries = yield* initEntriesContext();
 
+  let totalCost = {
+    cost: 0,
+    remaining: 0,
+    nodeCount: 0,
+    queryCount: 0,
+  };
+
   yield* spawn(function* () {
+    for (const item of yield* each(cost)) {
+      totalCost = {
+        cost: totalCost.cost + item.cost,
+        remaining: item.remaining,
+        nodeCount: totalCost.nodeCount + item.nodeCount,
+        queryCount: totalCost.queryCount + 1,
+      }
+      yield* each.next();
+    }
```

Comment:
You don't need a channel for this. You can create a closure function that accepts this data and updates internal state.